### DISCUSSION TITLE
aerosandVLM overprediction compared to XFLR5 and Star-CCM+

### DISCUSSION THREAD

**Discussion by harutidmore** [peterdsharpe/AeroSandbox#107] [Q&A] [answered]
Howdy,
I was running a validation case for my operating point before optimizing geometry and found that the VLM model in aerosandbox predicted a 5x5 (foot) S1223 at 4.5 m/s predicted around 15 lbs of lift at 4 aoa. 
In xflr 3d panels and horse shoe vortex predicted around 4 lbs
star-ccm+ using a rans model previously validated with wind tunnel data also predicted around 4 lbs...
Does anyone know if this is a direct cause of the solver or could this be how I set it up in aerosandbox.

**Comment by cdhainaut** [comment]
Hi,

There shouldn't be that much differences between XFLR or RANSE results with asb VLM or lifting line, especially in terms of lift which (at least at moderate CL) is linear and well captured with "low-fidelity" methods
Without knowing more or seeing your example it is difficult to tell what is wrong though.

I don't know exactly what you mean by a 5x5 foot S1223, but just some quick thoughts:

- Make sure you use same airfoils
- Compare indidual wing characteristics first. If the wing is not exagerately low aspect, you should find results that are in line with lifting line theory anyway (https://en.wikipedia.org/wiki/Lifting-line_theory check "useful approximations" paragraph)
- With XFLR you can check lift distribution and plot it vs aerosandbox to have more insights about what is going on

Bests,

**Comment by peterdsharpe** [comment]
Hi there,

@carlitador 's answer is spot-on.

One thing to consider is that a "5x5 foot" wing (aspect ratio 1) violates a lot of the embedded assumptions in a vortex-lattice formulation about spanwise vorticity convection on the wing surface - I'd be increasingly hesitant to trust any VLM method below aspect ratios of around 2 or so.

With that said, I wouldn't expect the deviance to be on the order of 3x. Would you be able to post a self-contained minimum-reproducible-example Python script of this analysis setup?

-----

Just napkin-math analyzing it, the numbers posted for the ASB VLM method would imply a lift coefficient of around 2.32 - not realistic at this angle of attack, even for a S1223.

One possibility is if the AeroSandbox wing is inadvertently symmetric (mirrored across the XZ plane), which would explain the somewhat-over-2x lift increase seen. Looking at the Python script of the example case will show whether or not this could be the case.

**Reply by harutidmore** [reply]
Howdy!
Here is what I have in python:

```
wing_airfoil = asb.Airfoil('S1223')

# Distances are in m and angles in deg

engl_b = 60
engl_c = 60
SI_b = engl_b/39.97
SI_c = engl_c /39.97

testsection = asb.Airplane(
    name = "2022 Wind Tunnel Model",
    xyz_ref = [0,0,0], #Cg
    wings=[
        asb.Wing(
            name ="MW",
            xyz_le = [0,0,0],
            symmetric= True,
            xsecs=[
                asb.WingXSec(
                    xyz_le =[0,0,0],
                    chord = SI_c,
                    twist = 0,
                    airfoil = wing_airfoil
                ),
                asb.WingXSec(
                    xyz_le =[0,SI_b/2,0],
                    chord = SI_c,
                    twist = 0,
                    airfoil = wing_airfoil
                )
            ]
        )
    ]
)

# Uncomment to visualize the geometry
testsection.draw_three_view()

#Running VLM
alpha = [-3.744,-2.376,-1.08,0.288,1.728,3.168,4.608]
log = []
VLM_cl = []
VLM_cd = []
for a in alpha:
    vlm = asb.VortexLatticeMethod(
        airplane= testsection,
        op_point = asb.OperatingPoint(
            velocity = 4.5,
            alpha = a
        )
    )
    aero = vlm.run()
    log.append(aero)
    VLM_cl.append(aero['CL'])
    VLM_cd.append(aero['CD'])
    print(aero["L"])
    print(VLM_cl)
```
Let me know if you need anything else.

**Reply by peterdsharpe** [reply, marked as answer]
Hmmm - when I run this code, I see 17.8 N (4.01 lbf) of lift at the last operating condition ($\alpha=4.608\degree$), which seems to agree well with other analysis:

Here's my IPython terminal:

```python
In[3]: runfile('C:\\Users\\peter\\AppData\\Roaming\\JetBrains\\PyCharm2023.2\\scratches\\scratch_138.py', wdir='C:\\Users\\peter\\AppData\\Roaming\\JetBrains\\PyCharm2023.2\\scratches')
11.465229315915813
[0.4102208402999755]
12.54578877947713
[0.4102208402999755, 0.4488827805823978]
13.558355314376707
[0.4102208402999755, 0.4488827805823978, 0.4851119639123397]
14.612866460258449
[0.4102208402999755, 0.4488827805823978, 0.4851119639123397, 0.5228419068947094]
15.704147351419033
[0.4102208402999755, 0.4488827805823978, 0.4851119639123397, 0.5228419068947094, 0.5618874551205753]
16.773252174740485
[0.4102208402999755, 0.4488827805823978, 0.4851119639123397, 0.5228419068947094, 0.5618874551205753, 0.6001395534351611]
17.817248484933348
[0.4102208402999755, 0.4488827805823978, 0.4851119639123397, 0.5228419068947094, 0.5618874551205753, 0.6001395534351611, 0.6374932802413816]
In[4]: vlm
Out[4]: 
VortexLatticeMethod(
	airplane=Airplane '2022 Wind Tunnel Model' (1 wing, 0 fuselages)
	op_point=OperatingPoint instance:
	State variables:
		atmosphere: Atmosphere (altitude: 0 m (0 ft), method: 'differentiable')                                                             
		  velocity: 4.5                                                                                                                     
		     alpha: 4.608                                                                                                                   
		      beta: 0.0                                                                                                                     
		         p: 0.0                                                                                                                     
		         q: 0.0                                                                                                                     
		         r: 0.0                                                                                                                     
	xyz_ref=[0 0 0]
)
In[5]: vlm.draw()
Out[5]: <pyvista.plotting.plotter.Plotter at 0x291cd4b83d0>
In[6]: import aerosandbox.tools.units as u
In[7]: aero["L"]
Out[7]: 17.817248484933348
In[8]: aero["L"] / u.lbf
Out[8]: 4.005477606995489
```

Is it possible that this is a Newtons vs. lbf unit issue on the output?

If not, perhaps it's the result of an old AeroSandbox version? Would you mind printing the results of `asb.__version__`?

**Comment by harutidmore** [comment]
Yes this is a units mistake on my part. Thanks for point out that as well
as the anticipated limitations of VLM.
Is there some documentation on the VLM model used?

Best
Haru
